src/appleDeviceUtils/appleDeviceUtils.js

```diff
@@ -0,0 +1,98 @@
+// Do not remove -- Apple devices change this when they support modern webkit messaging
+let hasModernWebkitAPI = false
+// INJECT hasModernWebkitAPI HERE
+
+// The native layer will inject a randomised secret here and use it to verify the origin
+const secret = 'PLACEHOLDER_SECRET'
+
+const ddgGlobals = window.navigator.ddgGlobals
+
+/**
+ * Sends message to the webkit layer (fire and forget)
+ * @param {String} handler
+ * @param {*} data
+ * @returns {*}
+ */
+const wkSend = (handler, data = {}) =>
+    window.webkit.messageHandlers[handler].postMessage(data)
+
+/**
+ * Generate a random method name and adds it to the global scope
+ * The native layer will use this method to send the response
+ * @param {String} randomMethodName
+ * @param {Function} callback
+ */
+const generateRandomMethod = (randomMethodName, callback) => {
+    Object.defineProperty(ddgGlobals.window, randomMethodName, {
+        enumerable: false,
+        // configurable, To allow for deletion later
+        configurable: true,
+        writable: false,
+        // Use proxy to ensure stringification isn't possible
+        value: new Proxy(function () {}, {
+            apply (target, thisArg, args) {
+                callback(...args)
+                delete ddgGlobals.window[randomMethodName]
+            }
+        })
+    })
+}
+
+/**
+ * Sends message to the webkit layer and waits for the specified response
+ * @param {String} handler
+ * @param {*} data
+ * @returns {Promise<*>}
+ */
+const wkSendAndWait = async (handler, data = {}) => {
+    if (hasModernWebkitAPI) {
+        return wkSend(handler, data)
+    }
+
+    const randMethodName = createRandMethodName()
+    const key = await createRandKey()
+    const iv = createRandIv()
+
+    const encryptedResponse = await new ddgGlobals.Promise((resolve) => {
+        generateRandomMethod(randMethodName, resolve)
+        data.messageHandling = {
+            methodName: randMethodName,
+            secret,
+            key: ddgGlobals.Arrayfrom(key),
+            iv: ddgGlobals.Arrayfrom(iv)
+        }
+        wkSend(handler, data)
+    })
+
+    return decrypt(encryptedResponse, key, iv)
+        .then(decrypted => ddgGlobals.JSONparse(decrypted))
+        .catch(e => { console.log(e); return {error: e} })
+}
+
+const randomString = () =>
+    '' + ddgGlobals.getRandomValues(new ddgGlobals.Uint32Array(1))[0] / 2 ** 32
```

Comment:
Drop the multiplication:
```suggestion
    '' + ddgGlobals.getRandomValues(new ddgGlobals.Uint32Array(1))[0]
```

Reply:
🤦 Thanks.